docs/howtoguides/trusty_legacy_support.rst

```diff
@@ -0,0 +1,105 @@
+How to enable esm-infra-legacy on Trusty
+*****************************************
+
+The ``esm-infra-legacy`` service can be enabled to grant additional two years of 
+extend security coverage. This service is only available on Trusty, since the release has reached
+its end of life support for esm-infra.
+
+
+Make sure ``ua`` is up-to-date
+===============================
+
+All systems come with ``ua`` pre-installed through the
+``ubuntu-advantage-tools`` package. To make sure that you're running the latest
+version of ``ua``, run the following commands:
+
+.. code-block:: bash
+
+    sudo apt update && sudo apt install ubuntu-advantage-tools
+
+
+Check the status of the services
+================================
+
+After you have attached to a Pro subscription and
+updated the ``ubuntu-advantage-tools`` package, you can check which services
+are enabled by running the following command:
+
+.. code-block:: bash
+
+    sudo ua status
+
+This will show you which services are enabled or disabled on your machine
+(output truncated for brevity):
+
+.. code-block:: text
+
+    SERVICE          ENTITLED  STATUS    DESCRIPTION
+    esm-infra        yes       enabled   Expanded Security Maintenance for Infrastructure
+    esm-infra-legacy yes       disabled  Expanded Security Maintenance for Infrastructure on Legacy Instances
+    livepatch        yes       enabled   Canonical Livepatch service
+
+If you are entitled to the service, you can enable it by running:
+
+
+.. code-block:: bash
+
+    sudo ua enable esm-infra-legacy
+
+
+What to do if esm-infra-legacy is not entitled
+===============================================
+
+If status is showing that you are not entitled to the service, you
+will need to buy access to the service. You can contact sales/support for help on that regard.
+
+Once you have bought support, you will need to run the following command
+on your machine:
+
+.. code-block:: bash
+
+    sudo ua refresh
+
+
+This command will refresh the contract definitions on your machine. After that, you
+can confirm that the service is now entitled by running:
+
+
+.. code-block:: bash
+
+    sudo ua status
+
+If you are now entitled to the service, you can enable it with:
+
+.. code-block:: bash
+
+    sudo ua enable esm-infra-legacy
+
+
+Trusty caveats
+===============
+
+There are some known caveats for the Trusty version of `ubuntu-advantage-tools`:
+
+* **Disabling esm-infra also disables esm-infra-legacy**: Even though this is happening, users
+  can disable esm-infra-legacy without consequences for esm-infra
+* **do-release-upgrade fails if esm-infra enabled and packages installed from service**: The
+  do-release-upgrade operation will fail here, as there will be an APT dependency chain issue
```

Comment:
1. remove "chain"
2. is this already true with esm-infra or only with esm-infra-legacy (the text suggests the former) - If it is the former it might need to go on a page not only talking about trusty-legacy?
3. Is there any workaround, IIRC there is not, but maybe I need to discuss the exact dependency issue that happens (I assume being ahead of the next release?). In any case we should give some helping hand - could be "Gladly that is rarely a real issue as production systems are usually deployed and rarely upgraded in place. The additional years provided by esm-infra and esm-infra-legacy are great to continue with what is known working and thereby extend the time you might need to set up a new production environment on a more recent Ubuntu LTS release."

Reply:
1. "packages installed from service" - what service?
2. I agree we need to provide some sort of instructions or guidance on how to work around the issue - at some point we want to encourage users to upgrade away from Trusty, but if do-release-update is going to break them then they're essentially stuck. We don't necessarily have to do that now/as part of this page, but we'll want to put something together to help.

Reply:
Yes, this is already happening with only `esm-infra`.  And as far as I know, there is no workaround here.
However, I don't know if we should use a separate page for it. What do you think @s-makin ?

Reply:
I think having the workaround in this page is fine :+1: